### DISCUSSION TITLE
How is everyone else going about endgames and checkmates?

### DISCUSSION THREAD

**Discussion by kittymuffin** [SebLague/Chess-Challenge#121] [Ideas]
I looked at Sebastian Lague's chess bot video but I don't get how he implemented checkmates. I tried to implement a similar thing that he did, but my bot won't do any checkmates at all :P. How is everyone else going about this?

**Comment by WhiteMouse1** [comment]
EvilBot has a a simple check if one of it's next moves will result in a checkmate against opponent and if there is such a move it picks it.
You can see it in Chess-Challenge/src/Evil Bot/EvilBot.cs

**Comment by draeg82** [comment]
board can return 2 booleans, isInCheck(), and isInCheckmate(), so when evaluating a move, check if isInCheckmate() is true and just go for that move.
